src/main/java/net/snowflake/client/core/json/Converters.java

```diff
@@ -71,4 +106,118 @@ public BytesConverter getBytesConverter() {
   public StringConverter getStringConverter() {
     return stringConverter;
   }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter integerConverter(int columnType) {
+    return value -> getNumberConverter().getInt(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter smallIntConverter(int columnType) {
+    return value -> getNumberConverter().getShort(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter tinyIntConverter(int columnType) {
+    return value -> getNumberConverter().getByte(value);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter bigIntConverter(int columnType) {
+    return value -> getNumberConverter().getBigInt(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter longConverter(int columnType) {
+    return value -> getNumberConverter().getLong(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter bigDecimalConverter(int columnType) {
+    return value -> getNumberConverter().getBigDecimal(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter floatConverter(int columnType) {
+    return value -> getNumberConverter().getBigDecimal(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter doubleConverter(int columnType) {
+    return value -> getNumberConverter().getBigDecimal(value, columnType);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter bytesConverter(int columnType, int scale) {
+    return value -> getBytesConverter().getBytes(value, columnType, Types.BINARY, scale);
+  }
+
+  @SnowflakeJdbcInternalApi
+  public JsonStringToTypeConverter varcharConverter(int columnType, int columnSubType, int scale) {
```

Comment:
Is it a varchar converter or the string one?

Reply:
return type is more important, so stringConverter.